scripts/Split_View_Figure.py

```diff
@@ -0,0 +1,262 @@
+
+    json_bytes = json.dumps(figure_json).encode('utf-8')
+    file_size = len(json_bytes)
+    f = BytesIO()
+    f.write(json_bytes)
+
+    update = conn.getUpdateService()
+    orig_file = conn.createOriginalFileFromFileObj(
+        f, '', figure_name, file_size, mimetype="application/json")
+    fa = FileAnnotationI()
+    fa.setFile(OriginalFileI(orig_file.getId(), False))
+    fa.setNs(rstring(JSON_FILEANN_NS))
+    desc = json.dumps(description)
+    fa.setDescription(rstring(desc))
+    fa = update.saveAndReturnObject(fa, conn.SERVICE_OPTS)
+    return fa.getId().getValue()
+
+def get_ch_label(image, ch_index):
+    channel = image.getChannels()[ch_index]
+    # positions are: top, left, right, leftvert, bottom, topleft, topright, bottomleft, bottomright
+    return {
+        "text": channel.getLabel(),
+        "size": 12,
+        "position": "top",
+        "color": channel.getColor().getHtml()
+    }
+
+def get_scalebar_json():
+    """Return JSON to add a 10 micron scalebar to bottom-right."""
+    return {"show": True,
+            "length": 10,
+            "units": "MICROMETER",
+            "position": "bottomright",
+            "color": "FFFFFF",
+            "show_label": True,
+            "font_size": 10}
+
+
+def get_panel_json(image, x, y, width, height, c_index=None):
+    """Get json for a figure panel."""
+    px = image.getPrimaryPixels().getPhysicalSizeX()
+    py = image.getPrimaryPixels().getPhysicalSizeY()
+
+    # channelMarshal gives us 'active':False for each channel
+    channels = [channelMarshal(x) for x in image.getChannels()]
+
+    # Just turn on 1 channel
+    if c_index is not None:
+        channels[c_index]['active'] = True
+    else:
+        # Or ALL channels (merged image)
+        for c in channels:
+            c['active'] = True
+
+    img_json = {
+        "imageId": image.id,
+        "y": y,
+        "x": x,
+        "width": width,
+        "height": height,
+        "orig_width": image.getSizeX(),
+        "orig_height": image.getSizeY(),
+        "sizeT": image.getSizeT(),
+        "sizeZ": image.getSizeZ(),
+        "channels": channels,
+        "name": image.getName(),
+        "theT": image.getDefaultT(),
+        "theZ": image.getDefaultZ(),
+    }
+    if px is not None:
+        img_json["pixel_size_x"] = px.getValue()
+        img_json["pixel_size_x_unit"] = str(px.getUnit())
+        img_json["pixel_size_x_symbol"] = px.getSymbol()
+    if py is not None:
+        img_json["pixel_size_y"] = py.getValue()
+    if image.getSizeT() > 1:
+        img_json['deltaT'] = get_timestamps(conn, image)
+    return img_json
+
+
+def create_omero_figure(conn, images):
+    """Create OMERO.figure from given images."""
+    figure_json = {"version": 5}
+
+    panel_width = 80
+    panel_height = panel_width
+    spacing = panel_width/20
+    margin = 40
+
+    panels_json = []
+
+    for i, image in enumerate(images):
+        print('image', image.id, image.name)
```

Comment:
do you want to keep that?

Reply:
Yes, I think it's useful for seeing progress, especially if an error occurred, you know which image it failed on.